homeassistant/components/vivotek/config_flow.py

```diff
@@ -0,0 +1,153 @@
+"""Config flow for Vivotek IP cameras integration."""
+
+import logging
+from typing import Any
+
+from libpyvivotek.vivotek import VivotekCameraError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import (
+    CONF_AUTHENTICATION,
+    CONF_IP_ADDRESS,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+    HTTP_BASIC_AUTHENTICATION,
+    HTTP_DIGEST_AUTHENTICATION,
+)
+from homeassistant.helpers import config_validation as cv
+
+from . import async_build_and_test_cam_client
+from .camera import DEFAULT_NAME, DEFAULT_SECURITY_LEVEL, DEFAULT_STREAM_SOURCE
+from .const import CONF_FRAMERATE, CONF_SECURITY_LEVEL, CONF_STREAM_PATH, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DESCRIPTION_PLACEHOLDERS = {
+    "doc_url": "https://www.home-assistant.io/integrations/vivotek/"
+}
+
+CONF_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): cv.string,
```

Comment:
We shouldn't ask for a name except for when its a helper. We should pick a name the user can recognize and then they can rename it via the UI